lte/gateway/python/integ_tests/s1aptests/test_attach_standalone_act_dflt_ber_ctxt_rej.py

```diff
@@ -0,0 +1,122 @@
+
+    def tearDown(self):
+        self._s1ap_wrapper.cleanup()
+
+    def test_attach_standalone_ActvDfltBearCtxtRej(self):
+        """ Test case for sending Activate Default
+        EPS Bearer Reject for secondary PDN """
+
+        self._s1ap_wrapper.configUEDevice(1)
+        req = self._s1ap_wrapper.ue_req
+
+        # APN of the secondary PDN
+        ims = {
+            "apn_name": "ims",  # APN-name
+            "qci": 5,  # qci
+            "priority": 15,  # priority
+            "pre_cap": 0,  # preemption-capability
+            "pre_vul": 0,  # preemption-vulnerability
+            "mbr_ul": 200000000,  # MBR UL
+            "mbr_dl": 100000000,  # MBR DL
+        }
+
+        # APN list to be configured
+        apn_list = [ims]
+
+        self._s1ap_wrapper.configAPN(
+            "IMSI" + "".join([str(i) for i in req.imsi]), apn_list
+        )
+        print(
+            "************************* Running End to End attach for UE id ",
+            req.ue_id,
+        )
+        # Attach
+        attach = self._s1ap_wrapper.s1_util.attach(
+            req.ue_id,
+            s1ap_types.tfwCmd.UE_END_TO_END_ATTACH_REQUEST,
+            s1ap_types.tfwCmd.UE_ATTACH_ACCEPT_IND,
+            s1ap_types.ueAttachAccept_t,
+        )
+
+        # Wait on EMM Information from MME
+        self._s1ap_wrapper._s1_util.receive_emm_info()
+
+        print("Sleeping for 5 seconds")
+        time.sleep(5)
+        # Trigger Activate Default EPS Bearer Context Reject indication
+        # so that s1ap tester sends Activate default EPS bearer context reject
+        # instead of Activate default EPS bearer context accept
+        def_ber_rej = s1ap_types.ueActvDfltEpsBearerCtxtRej_t()
+        def_ber_rej.ue_Id = req.ue_id
+        def_ber_rej.bearerId = attach.esmInfo.epsBearerId
+        def_ber_rej.cause = s1ap_types.TFW_EMM_CAUSE_PROT_ERR_UNSP
+
+        self._s1ap_wrapper._s1_util.issue_cmd(
+            s1ap_types.tfwCmd.
+            UE_STANDALONE_ACTV_DEFAULT_EPS_BEARER_CNTXT_REJECT,
+            def_ber_rej,
+        )
+
+        print(
+            "Sent STANDALONE_ACTV_DEFAULT_EPS_BEARER_CNTXT_REJECT indication"
+        )
+        print("Sleeping for 5 seconds")
+        time.sleep(5)
+        # Send PDN Connectivity Request
+        apn = "ims"
+        # PDN Type 1=IPv4, 2=IPv6, 3=IPv4v6
+        self._s1ap_wrapper.sendPdnConnectivityReq(req.ue_id, apn, pdn_type=3)
+        # Receive PDN CONN RSP/Activate default EPS bearer context request
+        response = self._s1ap_wrapper.s1_util.get_response()
+        self.assertEqual(
+            response.msg_type, s1ap_types.tfwCmd.UE_PDN_CONN_RSP_IND.value
+        )
+        act_def_bearer_req = response.cast(s1ap_types.uePdnConRsp_t)
+
+        print(
+            "************************* Sending Activate default EPS bearer "
+            "context reject for UE id %d and bearer %d"
+            % (req.ue_id, act_def_bearer_req.m.pdnInfo.epsBearerId)
+        )
```

Comment:
Are we sending anything? What is the purpose of the second PDN connection request?

Reply:
Can you move this print statement before line 92?

Reply:
disregard my previous comment. I got the answer in the other test case.